### PR TITLE
Backend notecard routing set up and basic page routing implemented

### PR DESCRIPTION
**Pull request by Rahmteen**
**Basic routing implementing using react-router-dom. Our hierarchy for the page structure is as follows:**

Screen Shot 2021-02-15 at 5 29 18 PM

- '/' _=> skeleton authorization page (waiting for functionality in future commits) and a designated skip button to '/home'_
- '/home' _=> the basic homepage that will allow users to add a notecard on visit to their desired deck._
- '/decks' _=> a repository of deck elements that represent the decks stored for each user, clicking on one prompts '/deck'_
- '/decks' _=> a page filled with the cards that were rendered based on the user input on '/decks'_

**Creation of Decks and Deck pages**

(DECK)
Screen Shot 2021-02-15 at 8 10 13 PM

(DECKS)
Screen Shot 2021-02-15 at 5 37 27 PM

- used the 'this.props.location.state' to access our state in the '/deck' page. This value is set in our code on the <Link> component. We are able to transfer this data with that functionality to the requested deck rendering page. 
- Also implemented Deck Routing with a Get request to our database.
- used the 'this.props.location.state' to access our state in the '/deck' page. This value is set in our code on the <Link> component. We are able to transfer this data with that functionality to the requested deck rendering page. 

**Addition of our temp authorization page is as follows:**
Screen Shot 2021-02-15 at 5 41 28 PM
- This serves as a landing page for our users and we will implement features that allow access to prior databases going forward.
- In the time being, there is a skip button to access the rest of the site for functionality and testing.

### CODE REVIEW

**Review by astaiglesia — APPROVED** [review]
Features addressed that have been verified:
- Client-side routing performs as expected
- GET request routed as noted with connection to back-end
- Authorization page appears; prepared for stretch feature of backend connection
- Components render as expected, displaying data returned by GET request

No exceptions are taken at this time and authorization of pull request is recommended

File: client/src/pages/Deck/Deck.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  //   const {deckNumber} = this.props.location.state
+
+  //   axios.get(`/deck/${deckNumber}`)
+  //     .then((res) => {
+  //       console.log("res: ", res);
+  //       this.setState({deck: res})
+  //     })
+  //     .catch((err) => {
+  //       console.log("err: ", err);
+  //   })
```
Comment (annieshinn) [inline client/src/pages/Deck/Deck.jsx]: Please edit your code to use a get request rather than a tester.

**Review by annieshinn — CHANGES_REQUESTED** [review]

File: client/src/pages/Deck/Deck.jsx (hunk 1 above)
Comment (Rahmteen) [reply client/src/pages/Deck/Deck.jsx]: The code has been updated to reflect a get request to our database as opposed to a test request.

**Review by annieshinn — APPROVED** [review]
Front-end team —

Thank you so much for implementing the changes that I suggested.

I reviewed the code after the implementation, and it seems to be as described:
> 1. There is a "home screen" landing page
> 2. There is a view of all the decks
> 3. Once a deck is clicked, the page is populated with Card components via the database

Awesome job. Keep up the good work!

**Review by taoantaoan — APPROVED** [review]
Awesome work! Way to realize the UI wireframes previously discussed. Thanks for taking our comments and criticisms into consideration when making your revisions. Very much appreciated the attention to detail. Here if you want to continue our brainstorming session on next steps for managing state.

**Comment by Rahmteen** [comment]
> Awesome work! Way to realize the UI wireframes previously discussed. Thanks for taking our comments and criticisms into consideration when making your revisions. Very much appreciated the attention to detail. Here if you want to continue our brainstorming session on next steps for managing state.

**Review by zbrucker — APPROVED** [review]
The team has verified that all features have been implemented as described in the pull request. 

- React and HTML components render as anticipated. 
- Event handlers are live and are correctly implementing the expected actions. 
- POST requests successfully update the database. 
- GET requests successfully read data from the database and render the expected components to the page.

**Pull request merged**